### ISSUE TITLE
--vars=file handling issue

### ISSUE THREAD

**Issue by ThomasCr** [OpenVPN/easy-rsa#257]
Hi,
I downloaded current release from GitHub. Problem is, that when I set EASYRSA_PKI in my vars file, it gets not applied/processed.

```
thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ grep EASYRSA_PKI vars-ca1 
set_var EASYRSA_PKI             "$PWD/../CA1-OVPN"
#set_var EASYRSA_TEMP_FILE      "$EASYRSA_PKI/extensions.temp"
# when undefined here, default behaviour is to look in $EASYRSA_PKI first, then
# EASYRSA_PKI or EASYRSA dir (in that order.) NOTE that this file is Easy-RSA

thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ ./easyrsa --vars=.\vars-ca1 init-pki

Using SSL: openssl OpenSSL 1.1.0g  2 Nov 2017

init-pki complete; you may now create a CA or requests.
Your newly created PKI dir is: /home/thomas/Certs/EasyRSA-3.0.5-lx/pki

```
When I use it like this, it works - but I am sure thats not how it should work

```
thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ EASYRSA_PKI=$PWD/../CA1-OVPN ./easyrsa --vars=.\vars-ca1 init-pki

Using SSL: openssl OpenSSL 1.1.0g  2 Nov 2017

init-pki complete; you may now create a CA or requests.
Your newly created PKI dir is: /home/thomas/Certs/EasyRSA-3.0.5-lx/../CA1-OVPN

```

Background is, that I want to handle multiple CAs with one easyrsa instance. Thats how I used it with old EasyRSA scripts on Windows and Linux...

**Comment by ThomasCr** [comment]
ok, I added some debug vars...

```
thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ diff -Nru easyrsa.orig easyrsa
--- easyrsa.orig        2018-09-15 06:21:19.000000000 +0200
+++ easyrsa     2018-10-18 16:27:04.508598812 +0200
@@ -1127,6 +1127,7 @@
        prog_vars="${0%/*}/vars"
        # set up PKI path
        pki_vars="${EASYRSA_PKI:-$PWD/pki}/vars"
+echo "DEBUG1: $EASYRSA_VARS_FILE"
 
        # command-line path:
        if [ -f "$EASYRSA_VARS_FILE" ]; then
@@ -1142,6 +1143,7 @@
                vars="$prog_vars"
        fi
 
+echo "DEBUG2: $vars"
        # If a vars file was located, source it
        # If $EASYRSA_NO_VARS is defined (not blank) this is skipped
        if [ -z "$EASYRSA_NO_VARS" ] && [ -n "$vars" ]; then
```

and this are the results:

```
thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ ./easyrsa --vars=vars-ca1
DEBUG1: vars-ca1
DEBUG2: vars-ca1
./easyrsa: 1153: .: vars-ca1: not found
thomas@nb:~/Certs/EasyRSA-3.0.5-lx$ ./easyrsa --vars=".\vars-ca1"
DEBUG1: .
         ars-ca1
DEBUG2: 

Easy-RSA 3 usage and overview
[...]
thomas@nbc:~/Certs/EasyRSA-3.0.5-lx$ ./easyrsa --vars=.\vars-ca1
DEBUG1: .vars-ca1
DEBUG2: 

Easy-RSA 3 usage and overview
[...]
```
Hope you can fix it :+1:

**Comment by ThomasCr** [comment]
it seams, that the script only accept the full path. But when you have a space in folder name - the script also breaks!
I temporally modified the script like this:
```
--- easyrsa.orig        2018-09-15 06:21:19.000000000 +0200
+++ easyrsa     2018-10-18 16:53:03.675418637 +0200
@@ -1148,7 +1148,7 @@
                #shellcheck disable=SC2034
                EASYRSA_CALLER=1 
                # shellcheck disable=SC1090
-               . "$vars"
+               . "./$vars"
                notice "\
 Note: using Easy-RSA configuration from: $vars"
        fi

```
and call it like this:

```
thomas@nbc:~/Certs/EasyRSA-3.0.5-lx$ ./easyrsa --vars=vars-ca1

Note: using Easy-RSA configuration from: vars-pemat

Easy-RSA 3 usage and overview

USAGE: easyrsa [options] COMMAND [command-options]

A list of commands is shown below. To get detailed usage and help for a
command, run:
  ./easyrsa help COMMAND

For a listing of options that can be supplied before the command, use:
  ./easyrsa help options

Here is the list of commands available with a short syntax reminder. Use the
'help' command above to get full usage details.

  init-pki
  build-ca [ cmd-opts ]
  gen-dh
  gen-req <filename_base> [ cmd-opts ]
  sign-req <type> <filename_base>
  build-client-full <filename_base> [ cmd-opts ]
  build-server-full <filename_base> [ cmd-opts ]
  revoke <filename_base>
  gen-crl
  update-db
  show-req <filename_base> [ cmd-opts ]
  show-cert <filename_base> [ cmd-opts ]
  import-req <request_file_path> <short_basename>
  export-p7 <filename_base> [ cmd-opts ]
  export-p12 <filename_base> [ cmd-opts ]
  set-rsa-pass <filename_base> [ cmd-opts ]
  set-ec-pass <filename_base> [ cmd-opts ]

DIRECTORY STATUS (commands would take effect on these locations)
  EASYRSA: .
      PKI: /home/thomas/Certs/EasyRSA-3.0.5-lx/../CA1-OVPN

```

**Comment by ecrist** [comment]
Looks like this was actually fixed way back in January of 2018 in commit 09e2c3b5.  Closing this issue.
